modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/option.kt

```diff
@@ -485,3 +488,28 @@ interface OptionUnalign : Unalign<ForOption>, OptionSemialign {
       }
     }
 }
+
+@extension
+interface OptionZip : Zip<ForOption>, OptionSemialign {
+  override fun <A, B> Kind<ForOption, A>.zip(other: Kind<ForOption, B>): Kind<ForOption, Tuple2<A, B>> =
+    when (val l = this.fix()) {
+      is None -> None
+      is Some -> when (val r = other.fix()) {
+        is None -> None
+        is Some -> Option.just(l.t toT r.t)
+      }
+    }
+}
```

Comment:
this is just applicative tupled isn't it? `this.fix().tupled(other.fix())` (don't even think we need fix here). `tupled` should be on either `Apply` or `Applicative` not sure